packages/core/src/defer/instructions.ts

```diff
@@ -371,7 +557,26 @@ export function ɵɵdeferOnTimer(delay: number) {
  * @codeGenApi
  */
 export function ɵɵdeferPrefetchOnTimer(delay: number) {
-  scheduleDelayedPrefetching(onTimer(delay));
+  scheduleDelayedPrefetching(onTimer(delay), DeferBlockTrigger.Timer);
+}
+
+/**
+ * Creates runtime data structures for the `on timer` hydrate trigger.
+ * @param delay Amount of time to wait before loading the content.
+ * @codeGenApi
+ */
+export function ɵɵdeferHydrateOnTimer(delay: number) {
+  const lView = getLView();
+  const tNode = getCurrentTNode()!;
+  const hydrateTriggers = getHydrateTriggers(getTDeferBlockDetails(getTView(), tNode));
```

Comment:
nit: we can pass TView and TNode directly to save some bytes:
```suggestion
    const hydrateTriggers = getHydrateTriggers(getTView(), tNode);
```